plugins/trendmicro_visionone/icon_trendmicro_visionone/connection/connection.py

```diff
@@ -0,0 +1,66 @@
+import insightconnect_plugin_runtime
+from insightconnect_plugin_runtime.exceptions import (
+    ConnectionTestException,
+    PluginException,
+)
+from .schema import ConnectionSchema, Input
+
+# Custom imports below
+import pytmv1
+
+
+class Connection(insightconnect_plugin_runtime.Connection):
+    def __init__(self):
+        super(self.__class__, self).__init__(input=ConnectionSchema())
+
+    def connect(self, params: dict = None):
+        """
+        Connection config params are supplied as a dict in
+        params or also accessible in self.parameters['key']
+
+        The following will setup the var to be accessed
+          self.blah = self.parameters['blah']
+        in the action and trigger files as:
+          blah = self.connection.blah
+        """
+        self.logger.info("Connect: Connecting...")
+
+        url = params.get(Input.API_URL, "")
+        if not url:
+            raise PluginException(
+                cause="No API URL provided.",
+                assistance="API URL is a required parameter, please provide a valid API URL.",
+                data=url,
+            )
```

Comment:
I checked and you have set `api_url` as a required field in the `plugin.spec.yaml`, so this check for url and raising PluginException is redundant.